share/ruby-install/functions.sh

```diff
@@ -19,7 +19,10 @@ function load_dependencies_from()
 {
 	local file="$1"
 
-	ruby_dependencies=($(fetch "$ruby/$file" "$package_manager" || return $?))
+	ruby_dependencies=()
+	while IFS='' read -r line; do
+		ruby_dependencies+=("$line")
+	done < <(fetch "$ruby/$file" "$package_manager" || return $?)
```

Comment:
We actually need the implicit shell-splitting here. `fetch "$ruby/$file" "$package_manager"` will read the `dependencies.txt` file and return the `apt: pkg1 pkg2 ...` line that matches the `"$package_manager"`, but without the `apt: ` prefix. It is then implicitly splatted into an Array of individual package names. Reading each line would cause `ruby_dependencies` to become a singleton Array that contains the whole line (ex: `("pkg1 pkg2 pkg3")`).

Reply:
Hmm, agree. I just followed (blindly) what Shellcheck WIKI was proposing :D